packages/xo-server/src/index.js

```diff
@@ -283,12 +283,14 @@ async function registerPlugin(pluginPath, pluginName) {
   } = plugin
   let instance
 
+  const config = this._config
   const handleFactory = factory =>
     typeof factory === 'function'
       ? factory({
+          config: config.plugins?.[pluginName],
```

Comment:
IMHO, `xo-server` should always pass a config, and defaults to an empty one if missing.